### ISSUE TITLE
Uninstalled module disappears from backoffice modules and marketplace section.

### ISSUE THREAD

**Issue by roja45** [PrestaShop/PrestaShop#33612]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

If you uninstall a module from the backoffice, it disappears completely even though the directory is still on the server.  

### Expected behavior

Previously you could find it in the marketplace section to reinstall.

### Steps to reproduce

1) Install a module by uploading a zip
2) Uninstall the module
3) Try to find it in the marketplace to reinstall

### PrestaShop version(s) where the bug happened

1.7.8.9

### PHP version(s) where the bug happened

7.4

**Comment by hibatallahAouadni** [comment]
Hello @roja45 

Could you please specify which module that you tried to install/uninstall :pray: 
Cause with the native modules I couldn't reproduce the issue.
Waiting for your feedback.

Thanks!

**Comment by roja45** [comment]
You can test with this module: 
[testtranslation-master-1.0.0.zip](https://github.com/PrestaShop/PrestaShop/files/12344972/testtranslation-master-1.0.0.zip)


Install the module from the zip
![image](https://github.com/PrestaShop/PrestaShop/assets/19314911/45bcad6b-ed46-44e8-9bdd-bb53612a4132)

Then uninstall it.

Previously, now that the files are on the filesystem, it would appear here, so that you can reinstall it:
![image](https://github.com/PrestaShop/PrestaShop/assets/19314911/934cf5b3-bf3c-49a8-86d8-7fef05d491a2)

But now it doesn't

Doesn't appear here either.
![image](https://github.com/PrestaShop/PrestaShop/assets/19314911/fab903fa-5266-4c86-8f3e-69da96a2b028)


Basically, the only way to install the module now is to use the zip.  From a dev point of view this is unusable.  Even from a customer point of view, if they have happened to change anything, it would be overwritten.

**Comment by roja45** [comment]
Where can I download v2.2.0 of the mbo module from?

**Comment by Thymotep** [comment]
Hello @roja45,

The problem seems to come from the ps_mbo module, which is not linked to the open-source project. You can find the old releases and open your issue directly on this project : https://github.com/PrestaShopCorp/ps_mbo/
